patches/proxy-uri.diff

```diff
@@ -115,22 +122,22 @@ Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
  import type { IURLCallbackProvider } from 'vs/workbench/services/url/browser/urlService';
  import type { IWorkbenchConstructionOptions } from 'vs/workbench/browser/web.api';
  import type { IWorkspace, IWorkspaceProvider } from 'vs/workbench/services/host/browser/browserHostService';
-+import { extractLocalHostUriMetaDataForPortMapping } from 'vs/platform/tunnel/common/tunnel';
++import { extractLocalHostUriMetaDataForPortMapping, TunnelOptions, TunnelCreationOptions } from 'vs/platform/tunnel/common/tunnel';
  
  interface ICredential {
  	service: string;
-@@ -511,6 +512,21 @@ function doCreateUri(path: string, query
+@@ -511,6 +512,39 @@ function doCreateUri(path: string, query
  		} : undefined,
  		workspaceProvider: WorkspaceProvider.create(config),
  		urlCallbackProvider: new LocalStorageURLCallbackProvider(config.callbackRoute),
 -		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider() // with a remote, we don't use a local credentials provider
 +		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider(), // with a remote, we don't use a local credentials provider
 +		resolveExternalUri: (uri: URI): Promise<URI> => {
 +			let resolvedUri = uri
-+			const localhostMatch = extractLocalHostUriMetaDataForPortMapping(uri)
++			const localhostMatch = extractLocalHostUriMetaDataForPortMapping(resolvedUri)
 +
 +			if (localhostMatch) {
-+				if (config.productConfiguration && config.productConfiguration.proxyEndpointTemplate) {
++				if ((resolvedUri.authority !== location.host) && config.productConfiguration && config.productConfiguration.proxyEndpointTemplate) {
```

Comment:
This will cause it to throw an error saying it is unable to resolve due to the product configuration being missing right?  We might need to move it up a bit.

```
  if (localhostMatch && resolvedUri.authority !== location.host) {
				if (config.productConfiguration && config.productConfiguration.proxyEndpointTemplate) {
```

Reply:
Good call. I'll move it up.